### DISCUSSION TITLE
Only 3 Rules? How to make Rules for 4 PIR Sensors

### DISCUSSION THREAD

**Discussion by mochoandre** [arendst/Tasmota#12356] [Q&A] [answered]
Hello,
I'm deploiyng a esp32 with 4 PIR and other "things". 
I've done 3 Rules for the 3 first PIR but when I try to do the 4th Rule for the last one it says 
`22:49:27.125 CMD: Rule4 on Switch4#state=1 do Backlog Publish stat/%topic%/PIR4 ON; RuleTimer4 30 endon on Rules#Timer=4 do Publish stat/%topic%/PIR4 OFF endon
22:49:27.138 MQT: stat/WC_ESP32/RESULT = {"Command":"Error"}`

Now I've read the wiki and it looks like there are only 3 Rules available. Is This Right? Not possible to use 4 PIR with corresponding Rules or how do I do it?

Regards and Thanks

**Comment by ascillato** [comment, marked as answer]
Tasmota supports 3 rule sets. Not 3 rules. You can have as much rules as you want on each slot until 511 characters without compression. Tasmota also compress the rules, so you can have many more.

For example 2 rules in one rule set.

Turn on rule set number 1
Rule1 1

Set the rules
Rule1 on switch1#state=2 do power1 2 endon on switch2#state do Var2 %value% endon

**Reply by ascillato** [reply]
For more information on how to use rules please search in the docs and also in the Tasmota support chat. Thanks.

**Reply by mochoandre** [reply]
Please, Is This correctly formated?
`Rule1 on Switch1#state=1 do Backlog Publish stat/%topic%/PIR1 ON; RuleTimer1 30 endon on Rules#Timer=1 do Publish stat/%topic%/PIR1 OFF endon on Switch2#state=1 do Backlog Publish stat/%topic%/PIR2 ON; RuleTimer2 30 endon on Rules#Timer=2 do Publish stat/%topic%/PIR2 OFF endon on Switch3#state=1 do Backlog Publish stat/%topic%/PIR3 ON; RuleTimer3 30 endon on Rules#Timer=3 do Publish stat/%topic%/PIR3 OFF endon on Switch4#state=1 do Backlog Publish stat/%topic%/PIR4 ON; RuleTimer4 30 endon on Rules#Timer=4 do Publish stat/%topic%/PIR4 OFF endon `

**Reply by mochoandre** [reply]
More then 511 Characters. So will it be?
`Rule1 on Switch1#state=1 do Backlog Publish stat/%topic%/PIR1 ON; RuleTimer1 30 endon on Rules#Timer=1 do Publish stat/%topic%/PIR1 OFF endon on Switch2#state=1 do Backlog Publish stat/%topic%/PIR2 ON; RuleTimer2 30 endon on Rules#Timer=2 do Publish stat/%topic%/PIR2 OFF endon`

`Rule2 on Switch3#state=1 do Backlog Publish stat/%topic%/PIR3 ON; RuleTimer3 30 endon on Rules#Timer=3 do Publish stat/%topic%/PIR3 OFF endon on Switch4#state=1 do Backlog Publish stat/%topic%/PIR4 ON; RuleTimer4 30 endon on Rules#Timer=4 do Publish stat/%topic%/PIR4 OFF endon`

**Reply by mochoandre** [reply]
Ok. Tryed and it works. Thanks you!!
